src/daft-dsl/src/expr/mod.rs

```diff
@@ -216,8 +216,11 @@ pub enum Expr {
         inputs: Vec<ExprRef>,
     },
 
+    #[display("{_0} over {_1}")]
+    Over(WindowExpr, window::WindowSpec),
+
     #[display("window({_0})")]
-    Window(ExprRef, window::WindowSpec),
+    Window(WindowExpr),
```

Comment:
Let's call this WindowFunction instead and add some comments here to talk about the purpose of Over vs WindowFunction